File: app/src/main/java/io/github/tobyhs/weatherweight/api/accuweather/AccuWeatherApiKeyInterceptor.java
package io.github.tobyhs.weatherweight.api.accuweather;

import java.io.IOException;

import okhttp3.HttpUrl;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * An OkHttp interceptor to add an API key to every AccuWeather API request
 */
public class AccuWeatherApiKeyInterceptor implements Interceptor {
    private final String apiKey;

    /**
     * @param apiKey AccuWeather API key
     */
    public AccuWeatherApiKeyInterceptor(String apiKey) {
        this.apiKey = apiKey;
    }

    @Override
    public Response intercept(Interceptor.Chain chain) throws IOException {
        Request originalRequest = chain.request();
        HttpUrl url = originalRequest.url().newBuilder().addQueryParameter("apikey", apiKey).build();
        Request newRequest = originalRequest.newBuilder().url(url).build();
        return chain.proceed(newRequest);
    }
}

File: app/src/main/java/io/github/tobyhs/weatherweight/api/accuweather/AccuWeatherApiKeyInterceptor.kt
package io.github.tobyhs.weatherweight.api.accuweather

import okhttp3.Interceptor
import okhttp3.Response

/**
 * An OkHttp interceptor to add an API key to every AccuWeather API request
 *
 * @param apiKey AccuWeather API key
 */
class AccuWeatherApiKeyInterceptor(private val apiKey: String) : Interceptor {
    override fun intercept(chain: Interceptor.Chain): Response {
        val originalRequest = chain.request()
        val url = originalRequest.url.newBuilder().addQueryParameter("apikey", apiKey).build()
        val newRequest = originalRequest.newBuilder().url(url).build()
        return chain.proceed(newRequest)
    }
}
